main.go

```diff
@@ -251,3 +326,19 @@ func rawSetuid(uid int) error {
 
 	return nil
 }
+
+func (handler *BuildHandler) saveNewBuild(repoName string) BuildInfo {
+	handler.buildListMutex.Lock()
+	defer handler.buildListMutex.Unlock()
+	info := BuildInfo{
```

Comment:
Now move next 6 lines to the 243 line and replace `saveNewBuild` call by structure initialization. Change signature of `saveNewBuild` so it will accept `info BuldInfo` instead of `repoName string`. Then, there will be no need for `pointer` field and `Save()` method. That was my point.